examples/image_segmentation_DIS/android/app/src/main/java/com/example/segmentationdis/ImageSegmentationHelper.kt

```diff
@@ -0,0 +1,185 @@
+ * limitations under the License.
+ */
+
+package com.example.segmentationdis
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.os.SystemClock
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.withContext
+import org.tensorflow.lite.DataType
+import org.tensorflow.lite.Interpreter
+import org.tensorflow.lite.support.common.FileUtil
+import org.tensorflow.lite.support.common.ops.NormalizeOp
+import org.tensorflow.lite.support.image.ImageProcessor
+import org.tensorflow.lite.support.image.TensorImage
+import org.tensorflow.lite.support.image.ops.ResizeOp
+import java.nio.FloatBuffer
+
+class ImageSegmentationHelper(private val context: Context) {
+    companion object {
+        private const val TAG = "SegmentationDis"
+    }
+
+    /** As the result of sound classification, this value emits map of probabilities */
+    val segmentation: SharedFlow<SegmentationResult>
+        get() = _segmentation
+    private val _segmentation = MutableSharedFlow<SegmentationResult>(
+        extraBufferCapacity = 64, onBufferOverflow = BufferOverflow.DROP_OLDEST
+    )
+
+    val error: SharedFlow<Throwable?>
+        get() = _error
+    private val _error = MutableSharedFlow<Throwable?>()
+
+    private var interpreter: Interpreter? = null
+
+    private var currentDelegate = Delegate.CPU
+    private var currentModel = Model.ISNET_TFL_DRQ
+
+    /** Init a Interpreter from Delegate and Model.*/
+    suspend fun initClassifier(
+        delegate: Delegate? = null,
+        model: Model? = null
+    ) {
+        delegate?.let { currentDelegate = it }
+        model?.let { currentModel = it }
+        interpreter = try {
+            val litertBuffer = FileUtil.loadMappedFile(context, currentModel.fileName)
+            val options = Interpreter.Options().apply {
+                numThreads = 4
+                useNNAPI = currentDelegate == Delegate.NNAPI
+            }
+            Interpreter(litertBuffer, options)
+        } catch (e: Exception) {
+            _error.emit(e)
+            null
+        }
+    }
+
+    /**
+     * Start read model and get bitmap result
+     */
+    suspend fun segment(bitmap: Bitmap) {
+        try {
+            val percent = bitmap.width.toFloat() / bitmap.height.toFloat()
+            withContext(Dispatchers.IO) {
+                if (interpreter == null) return@withContext
+                val startTime = SystemClock.uptimeMillis()
+
+                val (_, h, w, _) = interpreter?.getInputTensor(0)?.shape() ?: return@withContext
+                val imageProcessor =
+                    ImageProcessor.Builder()
+                        .add(ResizeOp(h, w, ResizeOp.ResizeMethod.BILINEAR))
```

Comment:
When resizing the image to fit on the screen, is there a way to sizes it so it isn't cropping?

Reply:
For example, I was playing with this using a different background to test different fabric colors for cross-stitching patterns. The image taken by the camera is cropped, zoomed, and off center when displayed, however.  https://photos.app.goo.gl/T3Yi3j131NxP4d4k8